Firmware/mmu2_protocol_logic.cpp

```diff
@@ -9,36 +9,43 @@
     // irrelevant on Buddy FW, just keep "_millis" as "millis"
     #include <wiring_time.h>
     #define _millis millis
+    #ifdef UNITTEST
+        #define strncmp_P strncmp
+    #else
+        #include <Marlin/src/core/serial.h>
+    #endif
 #endif
 
 #include <string.h>
 #include "mmu2_supported_version.h"
 
 namespace MMU2 {
 
-/// Beware:
+/// Beware - on AVR/MK3S:
 /// Changing the supportedMmuVersion numbers requires patching MSG_DESC_FW_UPDATE_NEEDED and all its related translations by hand.
 ///
 /// The message reads:
 ///   "MMU FW version is incompatible with printer FW.Update to version 2.1.9."
 ///
 /// Currently, this is not possible to perform automatically at compile time with the existing languages/translations infrastructure.
 /// To save space a "dumb" solution was chosen + a few static_assert checks in errors_list.h preventing the code from compiling when the string doesn't match.
+/// -----
+/// On Buddy FW we should improve the error screen to be able to print formatted strings
```

Comment:
Does this TODO need to be in the code? I feel it would be better to create a JIRA ticket for this as it only relates to the 32-bit.